tests/debugmalloc/debugmalloc.c

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#define _GNU_SOURCE
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+
+static void* a_ptr;
+static void* b_ptr;
+static void* c_ptr;
+
+void c()
+{
+    c_ptr = malloc(13);
+}
+
+void b()
+{
+    b_ptr = malloc(45);
+    c();
+}
+
+void a()
+{
+    a_ptr = malloc(99);
+    b();
+}
+
+size_t debug_malloc_check(bool print_allocations)
+{
+    const long SYS_myst_debug_malloc_check = 2021;
```

Comment:
I prefer not hardcoding the syscall number. We should include <syscallext.h> instead. See "tests/interrupt" as an example.

Reply:
I don't like hardcoding it either, but this is intended as a pattern for the user who will not have access to <syscallext.h>, which is not installed. Should we consider installing this file?

Reply:
I could find no way to define a function in a user app so this solution depends on making a syscall. We might have other situations and maybe we should consider a "universal syscall" for all future usages.